src/cnc/models/config/resource.py

```diff
@@ -142,6 +146,7 @@ class DatabaseResourceSettings(BaseResourceSettings):
         default=None,
     )
     use_db_proxy: Optional[bool] = True
+    public: Optional[bool] = True
```

Comment:
Seeing where this is used in the templates, it may not work in some scenarios. We're not accounting for things like lambda + rds - where if you set public to False we'd get an error because we're referencing an undeclared resource (the vpc).